Added content for StyleCop section (#8337)
kamranahmedse/developer-roadmap: src/data/roadmaps/aspnet-core/content/stylecop-rules@R7Qk5hsEIl9dspQXdaJAJ.md

# StyleCop Rules

StyleCop is a tool used for developers to standardize their code and ensure they all follow the same syntax principles. With StyleCop, one standard can be defined in a `stylecop.json` file and shared across your team so that each member has the same guidelines when formatting your code. Beyond a single project, StyleCop can also be added as an extension, so all of the projects on your IDE follow the same formatting rules, this is especially useful if your organization follows the same rule standards for all projects.

Visit the following resources to learn more:

> - [@opensource@GitHub: StyleCop GitHub official page](https://github.com/StyleCop/StyleCop)
> - [@opensource@GitLab: StyeleCop Analyzers,a more modern version of StyleCop](https://github.com/DotNetAnalyzers/StyleCopAnalyzers)
> - [@video@StyleCop set up tutorial ](https://www.youtube.com/watch?v=dmpOKmz3lPw)
> - [@article@Guide on how to start using it](https://blog.submain.com/stylecop-detailed-guide/)

Reviewer: ```suggestion
- [@opensource@StyleCop GitHub official page](https://github.com/StyleCop/StyleCop)
- [@opensource@StyeleCop Analyzers, a more modern version of StyleCop](https://github.com/DotNetAnalyzers/StyleCopAnalyzers)
- [@video@The StyleCop setup and Advantages](https://www.youtube.com/watch?v=dmpOKmz3lPw)
- [@article@ StyleCop: A Detailed Guide to Starting and Using It](https://blog.submain.com/stylecop-detailed-guide/)
```

# StyleCop Rules

> StyleCop is a tool used for developers to standardize their code and ensure they all follow the same syntax principles.
> With StyleCop, one standard can be defined in a stylecop.json file and shared across your team so that each member has the same guidelines when formatting your code. Beyond a single project, StyleCop can also be added as an extension, so all of the projects on your IDE follows the same formatting rules, this is specially useful if your organization follows the same rule standards for all projects.

Reviewer: ```suggestion
StyleCop is a tool used for developers to standardize their code and ensure they all follow the same syntax principles. With StyleCop, one standard can be defined in a `stylecop.json` file and shared across your team so that each member has the same guidelines when formatting your code. Beyond a single project, StyleCop can also be added as an extension, so all of the projects on your IDE follow the same formatting rules, this is especially useful if your organization follows the same rule standards for all projects.
```
